x-pack/plugins/alerting/common/rule_snooze_type.ts

```diff
@@ -0,0 +1,14 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+export type RuleSnooze = Array<{
+  startTime: string;
+  duration: number;
+  timeZone?: string;
+  id?: string;
+  rRule?: string;
```

Comment:
Maybe we can use the type form the library `rrule` here.

Reply:
I don't want to import `rrule` outside of `/server` for now. That increases the alerting plugin bundle size to the point where it doesn't pass CI, so I'd prefer to wait to deal with that until we have a critical reason for including `rrule` on the frontend.

Reply:
or we can just fork it for now

Reply:
I meant that we can just copy the interface, so our API is well defined

Reply:
The following came up when @XavierM and I were talking about timezones over Zoom. I think we should do what @XavierM just suggested where we don't support the complex `rRule` strings, and instead have a strongly typed object for `rRule` string which is equivalent to the type definition for the `rrule` library.

Reply:
This PR is already very large and duplicating the schema would be a lot of additional code, so I'm tracking this here: https://github.com/elastic/kibana/issues/132164